draft-ietf-keytrans-protocol.md

```diff
@@ -1805,6 +1772,91 @@ each label are not provided, given that these can be cached from the
+
+## Management
+
+With the Third-Party Management deployment mode, a third party is responsible
+for the majority of the work of storing and operating the Transparency Log,
+while the Service Operator serves mainly to enforce access control and
+authenticate the addition of new entries. All user queries specified in
+{{user-operations}} are initially sent by users directly to the Service Operator
+to be forwarded to the Third-Party Manager if they pass access control.
+
+While other operations are forwarded by the Service Operator unchanged,
+`UpdateRequest` structures are forwarded to the Third-Party Manager with the
+Service Operator's signature attached:
+
+~~~ tls-presentation
+struct {
+  UpdateRequest request;
+  opaque signature<0..2^16-1>;
+} ManagerUpdateRequest;
+~~~
+
+The signature is computed as described in {{update-format}}. The Service
+Operator MUST maintain its own records (independent of the Third-Party Manager)
+for the greatest version of each label for the purpose of producing this
+signature.
+
+## Auditing
+
+With the Third-party Auditing deployment mode, the Service Operator obtains
+signatures from a lightweight Third-Party Auditor attesting to the fact that the
+Service Operator is constructing the tree correctly. These signatures are
+provided to users along with the responses to their queries.
+
+Each entry added to the log tree has a corresponding `AuditorUpdate` structure
+that allows the Third-Party Auditor to verify the log entry against stored state
+from previous log entries:
+
+~~~ tls-presentation
+struct {
+  uint64 timestamp;
+
+  PrefixLeaf added<0..2^16-1>;
+  PrefixLeaf removed<0..2^16-1>;
+
+  PrefixProof proof;
+} AuditorUpdate;
+~~~
+
+The `timestamp` field contains the timestamp of the corresponding log entry. The
+`added` field contains the list of `PrefixLeaf` structures that were added to
+the prefix tree in the corresponding log entry. The `removed` field contains the
+list of `PrefixLeaf` structures that were removed from the prefix tree.
+
+The `proof` field contains a batch lookup proof in the previous log entry's
+prefix tree for all search keys referenced by `added` or `removed. The
+`proof.results` field contains the result of the search for each element of
+`added` in the order provided, followed by the result of the search for each
+element of `removed` in the order provided.
+
+An auditor processes a single `AuditorUpdate` by following these steps:
+
+1. Verify that `timestamp` is greater than or equal to the timestamp of the
+   previous log entry.
+2. Verify that the `PrefixSearchResult` provided in `proof` for each element of
+   `added` has a `result_type` of `nonInclusionParent` or `nonInclusionLeaf`.
+3. Verify that the `PrefixSearchResult` provided in `proof` for each element of
+   `removed` has a `result_type` of `inclusion`.
+4. For each element of `removed`, verify that, with the addition of the new log
+   entry, the prefix tree leaf will have been published in at least one
```

Comment:
Why "at least one"? Why not "the most recent"?

Reply:
It's possible that if we add a new log entry, that new log entry will be distinguished immediately. This happens for example if the RMW is 0. That would make it the most recent distinguished log entry. But we just removed this search key from it, so it's not in the "most recent" distinguished log entry